### DISCUSSION TITLE
How can I use Meson to create Cython libraries?

### DISCUSSION THREAD

**Discussion by dreamer-coding** [mesonbuild/meson#13082] [Q&A] [answered]
How can I use Meson to create Cython libraries? Can this work with C and or C++? Is the Cython support only intended for libraries or can applications be built with Cython and Meson build?

**Comment by dcbaker** [comment]
You need to add the `cython` language to the `project()` call. You can control whether Cython generates C or C++ with the `cython_language` option, which can be either 'c' (the deafult) or 'cpp'. I think you can use it for executabled, but I'm not 100% sure.

**Comment by eli-schwartz** [comment, marked as answer]
Meson doesn't care how you use cython, it just provides a wrapper UX for listing cython sources and transpiling them to C or C++.

Cython itself supports embedding into a C/C++ program via the aptly named `cython --embed` which generates a main() function for you. Or you can write your own main() and simply link to the cython files. Either way it's still as simple as adding .pyx files as source files to your executable().
